pkgs/development/python-modules/torch/default.nix

```diff
@@ -227,6 +279,9 @@ in buildPythonPackage rec {
     mkdir $lib
     mv $out/${python.sitePackages}/torch/lib $lib/lib
     ln -s $lib/lib $out/${python.sitePackages}/torch/lib
+  '' + lib.optionalString rocmSupport ''
+    # We need this for the import check
+    export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:${rocmtoolkit_joined}/lib
```

Comment:
There has to be a better way, since ``torchrun`` also needs this.
Also getting something about ``typing_extensions`` not being found, which shouldn't be the case since it's a propagated build input.

Reply:
Perhaps this is relevant?
```console
python3.10-torch> /build/source/dist /build/source
python3.10-torch> Processing ./torch-1.13.0-cp310-cp310-linux_x86_64.whl
python3.10-torch> Requirement already satisfied: typing-extensions in /nix/store/b5zpk8l12yvqfgsb79z6gbibf8g4ny17-python3.10-typing-extensions-4.3.0/lib/python3.10/site-packages (from torch==1.13.0) (4.3.0)
python3.10-torch> Installing collected packages: torch
python3.10-torch> Successfully installed torch-1.13.0
python3.10-torch> /build/source
python3.10-torch> Finished executing pipInstallPhase
```

Reply:
Relevant: https://github.com/pytorch/pytorch/issues/71360
Will try with typing-extensions 4.4.0 once I rebuild the world.

Reply:
After plenty of testing, the ``typing-extensions`` import bug is also present in regular torch.
This seems to be caused by a change in python versions, and I have been unsuccessful on using a different python version of ``typing-extensions`` with torch.
Unfortunately, 4.4.0 does not help in the slightest.
At any rate, this isn't a problem exclusive to adding rocm at least.

Reply:
Removing ``typing-extensions`` from ``propagatedBuildInputs`` and linking it's site-packages contents to torch's site-packages works.